homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+
+    server = hass.data[AGENT_DOMAIN][config_entry.entry_id][CONNECTION]
+    server.cameras = []
+    if not server.deviceList:
+        _LOGGER.warning("Could not fetch cameras from Agent server")
+        return
+
+    for device in server.deviceList:
+        if device.typeID == 2:
+            camera = AgentCamera(device)
+            cameras.append(camera)
+            server.cameras.append(camera)
+
+    async_add_entities(cameras)
+
+    platform = entity_platform.current_platform.get()
+    for service, method in CAMERA_SERVICES.items():
+        platform.async_register_entity_service(service, {}, method)
+
+    return True
+
+
+class AgentCamera(MjpegCamera):
+    """Representation of an Agent Device Stream."""
+
+    def __init__(self, device, enabled_default: bool = True):
+        """Initialize as a subclass of MjpegCamera."""
+        self._servername = device.client.name
+        self.server_url = device.client._server_url
+
+        device_info = {
+            CONF_NAME: device.name,
+            CONF_MJPEG_URL: f"{self.server_url}{device.mjpeg_image_url}&size=640x480",
+            CONF_STILL_IMAGE_URL: f"{self.server_url}{device.still_image_url}&size=640x480",
+        }
+        self.device = device
+        self._removed = False
+        self._name = f"{self._servername} {device.name}"
+        self._stream_url = f"{self.server_url}{device.mp4_url}"
+        self._unique_id = f"{device._client.unique}_{device.typeID}_{device.id}"
+        self._enabled_default = enabled_default
+        super().__init__(device_info)
+
+    @property
+    def device_info(self):
+        """Return the device info for adding the entity to the agent object."""
+        return {
+            "identifiers": {(AGENT_DOMAIN, self._unique_id)},
+            "name": self._name,
+            "manufacturer": "Agent",
+            "model": "Camera",
+            "sw_version": self.device.client.version,
+        }
+
+    async def async_update(self):
+        """Update our state from the Agent API."""
+        try:
+            await self.device.update()
+            if self._removed:
+                _LOGGER.error("%s reacquired", self._name)
+            self._removed = False
+        except AgentError:
+            if self.device.client.is_available:  # server still available - camera error
+                if not self._removed:
+                    _LOGGER.error("%s lost", self._name)
+                    self._removed = True
+
+    @property
+    def state_attributes(self):
+        """Return the Agent DVR camera state attributes."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "editable": False,
+            "enabled": self.enabled,
+            "connected": self.connected,
+            "detected": self.is_detected,
+            "alerted": self.is_alerted,
+            "recording": self.is_recording,
+            "has_ptz": self.device.has_ptz,
+            "motion_detection_enabled": self.device.detector_active,
```

Comment:
`"motion_detection"` is already available in the camera base state attributes.

Reply:
Agent does it's own motion detection

Reply:
i'll remove all the agent motion detection functionality

Reply:
Implement the `motion_detection_enabled` property instead.
https://github.com/home-assistant/core/blob/1b3f9254bc5b392c29a3219bb1b82d127036d18c/homeassistant/components/camera/__init__.py#L351-L353

```suggestion
```

Reply:
Bump.